io-engine/src/core/logical_volume.rs

```diff
@@ -1,7 +1,10 @@
 use crate::lvs::LvolSpaceUsage;
+use spdk_rs::libspdk::spdk_lvol;
 
 ///  LogicalVolume Trait Provide all the Generic Interface for Volume
 pub trait LogicalVolume {
+    /// Get lvol inner ptr.
+    fn as_inner_ptr(&self) -> *mut spdk_lvol;
```

Comment:
hmm this is kinda breaking the api...
Maybe we can have an associated type here?

Reply:
I kinda agree with Tiago: trait should hide low-level implementation and expose an abstract logical volume, rather than give details about spdk-based lvol implementation.

Reply:
Got it. Will change it accordingly.